src/hotspot/share/opto/convertnode.cpp

```diff
@@ -165,17 +166,14 @@ const Type* ConvF2DNode::Value(PhaseGVN* phase) const {
 //------------------------------Value------------------------------------------
 const Type* ConvF2HFNode::Value(PhaseGVN* phase) const {
   const Type *t = phase->type( in(1) );
-  if( t == Type::TOP ) return Type::TOP;
-  if( t == Type::FLOAT ) return TypeInt::SHORT;
+  if (t == Type::TOP) return Type::TOP;
+  if (t == Type::FLOAT) return TypeInt::SHORT;
+  if (StubRoutines::f2hf() == nullptr) return bottom_type();
```

Comment:
What's the purpose of this check? My understanding is ConvF2HF/ConvHF2F require intrinsification and on platforms where stubs are absent, intrinsification is disabled.

Reply:
Thanks for the clarifications. Now it makes much more sense.

Still, the mix of `StubRoutines::f2hf()` and `SharedRuntime::f2hf()` looks a bit confusing.

What if you move the wrapper to `StubRoutines` class instead? (`JRT_LEAF` et al stuff looks redundant here. Also, even though there are other arithmetic operations declared on `StubRoutines`, they provide default implementations universally available across all platforms. `f2hf` case is different since it exposes a platform-specific stub and its availability is limited.)  

Or encapsulate the constant folding logic (along with the guard) into `SharedRuntime` and return `Type*` (instead of int/float scalar).

Reply:
> Or encapsulate the constant folding logic (along with the guard) into SharedRuntime and return Type* (instead of int/float scalar).

I take this particular suggestion back. `SharedRuntime` is compiler-agnostic while `Type` is C2-specific.